tests/Doctrine/Migrations/Tests/Generator/ClassNameGeneratorTest.php

```diff
@@ -7,12 +7,19 @@
 use Doctrine\Migrations\Generator\ClassNameGenerator;
 use PHPUnit\Framework\TestCase;
 
+use function method_exists;
+
 class ClassNameGeneratorTest extends TestCase
 {
     public function testName(): void
     {
         $generator = new ClassNameGenerator();
         $fqcn      = $generator->generateClassName('Foo');
-        self::assertRegExp('/^Foo\\\\Version[0-9]{14}$/', $fqcn);
+
+        if (method_exists($this, 'assertMatchesRegularExpression')) {
+            $this->assertMatchesRegularExpression('/^Foo\\\\Version[0-9]{14}$/', $fqcn);
```

Comment:
I guess we should use the static implementation, so `self`

Reply:
That doesn't play well with phpstan

Reply:
More info:  `method_exists` cannot take `self` as the first parameter.  So if you use `$this` for `method_exists` then use `self` for `assertMatchesRegularExpression` `phpstan` has a problem with that.

Reply:
What about using the class name (can't use `$this::class` for now since it's PHP 8 but `TestCase::class` should do the trick)

Reply:
Nope.  You see, phpstan is `clearing` the use of $this after the method_exists function and seeing that it won't be called passes.  

```
        if (method_exists(TestCase::class, 'assertMatchesRegularExpression')) {
            self::assertMatchesRegularExpression('/^Foo\\\\Version[0-9]{14}$/', $fqcn);
```

Results in

```
 ------ ----------------------------------------------------------------------------------------------------------------------------------
  Line   tests/Doctrine/Migrations/Tests/Generator/ClassNameGeneratorTest.php
 ------ ----------------------------------------------------------------------------------------------------------------------------------
  20     Call to an undefined static method Doctrine\Migrations\Tests\Generator\ClassNameGeneratorTest::assertMatchesRegularExpression().
 ------ ----------------------------------------------------------------------------------------------------------------------------------
```